src/python/pants/engine/build_files.py

```diff
@@ -243,13 +243,23 @@ def addresses_from_address_families(address_families, spec):
   if not address_families:
     raise ResolveError('Path "{}" contains no BUILD files.'.format(spec.directory))
 
+  def exclude_address(address):
```

Comment:
@baroquebobcat : That filters at the wrong level (it's how this was implemented before): we don't want to prevent the target from being loaded at all, we just want to filter it out of roots (Specs represent roots).